x-pack/legacy/plugins/apm/public/components/shared/TimestampTooltip/index.tsx

```diff
@@ -12,36 +12,51 @@ interface Props {
    * timestamp in milliseconds
    */
   time: number;
+  endTime?: number;
   precision?: 'days' | 'minutes' | 'seconds' | 'milliseconds';
 }
 
-function getPreciseTime(precision: Props['precision']) {
+function getPreciseTime(
+  precision: Props['precision'],
+  separator: string = ','
+) {
   switch (precision) {
     case 'days':
       return '';
     case 'minutes':
-      return ', HH:mm';
+      return `${separator} HH:mm`;
     case 'seconds':
-      return ', HH:mm:ss';
+      return `${separator} HH:mm:ss`;
     default:
-      return ', HH:mm:ss.SSS';
+      return `${separator} HH:mm:ss.SSS`;
   }
 }
 
 function withLeadingPlus(value: number) {
   return value > 0 ? `+${value}` : value;
 }
 
-export function asAbsoluteTime({ time, precision = 'milliseconds' }: Props) {
+export function asAbsoluteTime({
+  time,
+  endTime,
+  precision = 'milliseconds'
+}: Props) {
   const momentTime = moment(time);
   const utcOffsetHours = momentTime.utcOffset() / 60;
   const utcOffsetFormatted = Number.isInteger(utcOffsetHours)
     ? withLeadingPlus(utcOffsetHours)
     : 'Z';
 
-  return momentTime.format(
-    `MMM D, YYYY${getPreciseTime(precision)} (UTC${utcOffsetFormatted})`
-  );
+  let format = `MMM D, YYYY${getPreciseTime(precision)}`;
+
+  if (endTime) {
+    const formattedEndTime = moment(endTime).format(
+      getPreciseTime(precision, '-')
+    );
+    format = `${format} ${formattedEndTime}`;
+  }
+
+  return momentTime.format(`${format} (UTC${utcOffsetFormatted})`);
```

Comment:
What do you think about separating the formatting of ranges and single timestamps a little more:

```ts
export function asAbsoluteTime({
  time,
  endTime,
  precision = 'milliseconds'
}: Props) {
  const momentTime = moment(time);
  const utcOffsetHours = momentTime.utcOffset() / 60;
  const utcOffsetFormatted = Number.isInteger(utcOffsetHours)
    ? withLeadingPlus(utcOffsetHours)
    : 'Z';

  const formattedStartTime = momentTime.format(
    `MMM D, YYYY${getPreciseTime({ precision })}`
  );

  const formattedTz = momentTime.format(`(UTC${utcOffsetFormatted})`);

  // format as range
  if (endTime) {
    const formattedEndTime = moment(endTime).format(
      getPreciseTime({ precision, hasSeparator: false })
    );
    return `${formattedStartTime} - ${formattedEndTime} ${formattedTz}`;
  }

  // format as single timestamp
  return `${formattedStartTime} ${formattedTz}`;
}
```